letsencrypt-auto-source/letsencrypt-auto.template

```diff
@@ -462,6 +474,10 @@ if [ "$1" = "--le-auto-phase2" ]; then
         error "install any required packages."
         # Set INSTALLED_VERSION to be the same so we don't update the venv
         INSTALLED_VERSION="$LE_AUTO_VERSION"
+        # Continue to use OLD_VENV_PATH if the new venv doesn't exist
+        if [ ! -d "$VENV_PATH" ]; then
+          VENV_BIN="$OLD_VENV_PATH/bin"
```

Comment:
It's not like it's happening suddenly with this PR, but this whole script is becoming very brittle, with state changing everywhere and all of it global. It is difficult to reason about. And it's error prone: sure, we update VENV_BIN here, but we don't update VENV_PATH. But that's okay because all the subsequent uses of it are wrapped in `if [ "$LE_AUTO_VERSION" != "$INSTALLED_VERSION" ]`, which should always be true in this case. *For now.* The assumptions are unstated and invisible, and they keep piling up. As we just talked about on IRC, we should do something about that.

And I *think* VENV_PATH is intended to mean "The modern venv path, as opposed to the OLD_VENV_PATH" vs. "The venv that we're going to run stuff from." I think we should comment that explicitly where VENV_PATH is first initialized, in this PR. That will help hold the semantics together until we do a cleanup job.

Reply:
`VENV_PATH` normally means both of those things:

* The modern venv path, as opposed to the OLD_VENV_PATH
* The venv that we're going to run stuff from

But this PR changes that to allow us to continue to use the old venv.

I can add comments about this to the top of the script where these variables are set if you'd like.

Reply:
I move we retitle #4947 with your excellent observation: "Most people don't feel comfortable reviewing this." ;-)